### DISCUSSION TITLE
Sendgrid integration

### DISCUSSION THREAD

**Discussion by strstnts** [outline/outline#2589] [Self Hosting]
Hi! I set up sending mail for Outline (self-hosted). Unfortunately, sending emails does not work in principle, the following error in the logs (Sendgrid relay smtp tested on other clients– it's work):

```
outline_1   | 2021-09-16T14:25:40.150Z emails Sending email "*** invited you to join *** knowledge base" to ****@***.**
outline_1   | (node:71) UnhandledPromiseRejectionWarning: Error: Connection timeout
outline_1   |     at SMTPConnection._formatError (/opt/outline/node_modules/nodemailer/lib/smtp-connection/index.js:774:19)
outline_1   |     at SMTPConnection._onError (/opt/outline/node_modules/nodemailer/lib/smtp-connection/index.js:760:20)
outline_1   |     at Timeout.<anonymous> (/opt/outline/node_modules/nodemailer/lib/smtp-connection/index.js:229:22)
outline_1   |     at listOnTimeout (internal/timers.js:557:17)
outline_1   |     at processTimers (internal/timers.js:500:7)
outline_1   | (Use `node --trace-warnings ...` to show where the warning was created)
outline_1   | (node:71) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
outline_1   | (node:71) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

env.outline

```
SMTP_HOST=smtp.sendgrid.net
SMTP_PORT=587
SMTP_USERNAME=apikey
SMTP_PASSWORD=SG.andothercharsofsendgridpassword
SMTP_FROM_EMAIL=hereisvalidmail@example.com
SMTP_REPLY_EMAIL=hereisvalidmail@example.com
```

Outline available via Nginx reverse proxy (with Letsencrypt certificate). Help me, please 🥺

**Comment by jfmontanaro** [comment]
Have you tried port 465 instead of 587? Often I find when both are available, one will work with a given client and one won't. I think it depends on whether the client is set up to open a TLS connection from the very start, or start in plaintext and then upgrade using the STARTTLS command.

**Reply by strstnts** [reply]
Same error with port 465

**Reply by tommoor** [reply]
With the error being connection timeout it suggests that the container cannot reach the sendgrid server. I'm not aware of any common mistakes that would cause this but I trust you haven't restricted outgoing connections?

**Reply by strstnts** [reply]
I am using Hetzner Cloud, there might be problems with it? Should I somehow check my outgoing connections? Mail sending works with other scripts (e.g. PHPMailer and Nextcloud's mail client)...
